beacon_node/lighthouse_network/src/discovery/mod.rs

```diff
@@ -1156,8 +1156,19 @@ impl<E: EthSpec> Discovery<E> {
     fn on_dial_failure(&mut self, peer_id: Option<PeerId>, error: &DialError) {
         if let Some(peer_id) = peer_id {
             match error {
+                DialError::Denied { .. } => {
+                    if self.network_globals.peers.read().is_connected(&peer_id) {
+                        // There's an active connection, so we don’t disconnect the peer.
+                        // Lighthouse dials to a peer twice using TCP and QUIC (if QUIC is not
+                        // disabled). Usually, one establishes a connection, and the other fails
+                        // because the peer allows only one connection per peer.
+                        return;
+                    }
+                    // set peer as disconnected in discovery DHT
+                    debug!(self.log, "Marking peer disconnected in DHT"; "peer_id" => %peer_id, "error" => %ClearDialError(error));
+                    self.disconnect_peer(&peer_id);
```

Comment:
```suggestion
```
this seems redundant right? if the peer is not connected, we shouldn't need to disconnected it

Reply:
@age do you know why are we disconnecting from discovery on a dial failure instead of when a connection has been closed?

Reply:
In this case we are connected via discovery on UDP. However we cannot connect via libp2p. This could mean that the ENR they are advertising is not correct, so we want to remove them from our DHT so we don't advertise to others, because if we can't dial them, no one will be able to.

Reply:
yeah makes sense age, thanks!